src/main/java/seedu/address/model/tag/DeadlineTag.java

```diff
@@ -0,0 +1,60 @@
+package seedu.address.model.tag;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Deadline Tag represents a tag which contains the deadline for the task
+ */
+public class DeadlineTag {
+    public static final String DEADLINE_TAG_CONSTRAINTS =
+            "Deadline of the tag should be in the format dd-mm-yyyy and";
+    private static final DateTimeFormatter dtf = DateTimeFormatter.ofPattern("dd-MM-yyyy");
+
+    public final LocalDate deadline;
+
+    /**
+     * Constructor of the DeadlineTag. Sets the deadline of the
+     * tag.
+     *
+     * @param deadline The deadline of the tag.
+     */
+    public DeadlineTag(LocalDate deadline) {
+        requireNonNull(deadline);
+        checkArgument(isValidDeadline(deadline), DEADLINE_TAG_CONSTRAINTS);
+        this.deadline = deadline;
+    }
+
+    /**
+     * Checks whether the deadline given is valid.
+     *
+     * @param testDateAdded The date being checked for validity.
+     * @return true if the date is valid; else return false.
+     */
+    public static boolean isValidDeadline(LocalDate testDateAdded) {
```

Comment:
When this method is called from JsonAdaptedTask::toModelType, if the user forgets to delete a deadline that has already passed, it would throw an IllegalValueException when the app just starts up. Might have to find a way to overcome this.

Reply:
This is a valid bug, I will maybe add tie the date that the deadline was added to the DeadlineTag.

Reply:
I have resolved this issue by removing the check since any date should be allowed. It still checks for whether the date parsed is valid format before that so no subsequent issues will crop up as well.